tests/unit/src/utils.test.js

```diff
@@ -0,0 +1,86 @@
+import { randomString, safeSlot } from '../../../src/utils.js';
+
+describe('Utilities', () => {
+  describe('randomString', () => {
+    test('No inputs', () => {
+      const output = randomString();
+
+      expect(typeof(output))
+        .toEqual('string');
+
+      expect(output.length)
+        .toEqual(10);
+    });
+
+    test('Bad inputs', () => {
+      expect(randomString(NaN, {}))
+        .toEqual('');
+    });
+
+    test('Length 0', () => {
+      expect(randomString(0))
+        .toEqual('');
+    });
+
+    test('Length 1', () => {
+      const output = randomString(1);
+
+      expect(typeof(output))
+        .toEqual('string');
+
+      expect(output.length)
+        .toEqual(1);
+    });
+
+    test('Length 100', () => {
+      const output = randomString(100);
+
+      expect(typeof(output))
+        .toEqual('string');
+
+      expect(output.length)
+        .toEqual(100);
+    });
+
+    test('Only allow numbers', () => {
+      const allowed = '0123456789';
+      const output = randomString(10, allowed);
+
+      const filtered = output.split('').filter(function (character) {
+        return allowed.includes(character);
+      });
+
+      expect(output.length)
+        .toEqual(filtered.length);
+    });
+  });
+
+  describe('safeSlot', () => {
+    const render = function (element, slots) {
+      let markup = '';
+      slots.forEach(function (slot) {
+        markup = markup + '<' + element + '>' + slot + '</' + element + '>';
+      });
+      return markup;
+    };
+
+    test('No inputs', () => {
+      expect(safeSlot())
+        .toEqual(undefined);
+    });
+
+    test('One slot', () => {
+      const slot = ['a'];
+
+      expect(safeSlot(render, slot))
+        .toEqual(['a'])
+    });
+
+    test('Multiple slots', () => {
+      const slot = ['a', 'b'];
+
+      expect(safeSlot(render, slot))
+        .toEqual('<div>a</div><div>b</div>');
```

Comment:
Not really sure what the point of `safeSlot` is and why it is different depending on number of slots. So these tests will likely need improved.

Reply:
`safeSlot` is a simple util I have for checking how many root nodes are inside the slot. Vue only let's us put 1 root node, so if a developer places more than 1, I just wrap them in a `<div>`